gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/annotation/PushSubscription.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gcp.pubsub.annotation;
+
+import io.micronaut.context.annotation.AliasFor;
+import io.micronaut.context.annotation.Executable;
+import io.micronaut.messaging.annotation.MessageMapping;
+
+import java.lang.annotation.*;
+
+/**
+ * Represents a PubSub Push subscription. Methods annotated with this annotation
+ * will be invoked when a new message is received by a {@link com.google.cloud.pubsub.v1.MessageReceiver}
+ * that is bound to the annotated method.
+ *
+ * @author Jeremy Grelle
+ * @since 5.4.0
```

Comment:
We should bump `gradle.properties` so the version is `5.4.0-SNAPSHOT`

Reply:
Our version management process is still a little opaque to me. Should I do that as part of this PR, or is the process supposed to be that we'd do separately that in master before merging this?

Reply:
There's not a unified process that I know of 🤔 Either would work I guess, though doing it here is arguably less error prone?

Reply:
Ok, cool, went ahead and bumped it here. https://github.com/micronaut-projects/micronaut-gcp/pull/1012/commits/415f586180525872489f61cb73526963904d5828